src/transactions/OperationFrame.h

```diff
@@ -84,5 +84,8 @@ class OperationFrame
     {
         return mOperation;
     }
+
+    virtual std::unordered_set<LedgerKey>
+    getLedgerKeysToPrefetch(Application& app);
```

Comment:
I don't think we should include `app` here. @vogel is already working to clean this up in other places, see #1924.

Also one concern that I have about this interface is that it might lead to the construction of many intermediate containers. I'm not sure if it's worth changing this right now (I think this interface is much cleaner than one which would take `std::unordered_set<LedgerKey>` by reference for example), but it's something we should keep an eye on for performance reasons.

Reply:
Good ideas, agreed.